[swxk19] iP

swxk19 opened the pull request:
# Duke (Beta)

> "Don't lose track of your tasks! Your memory isn't perfect, but Duke's is! 👍" - [Me](https://github.com/swxk19)

**<ins>Why use Duke?:</ins>**

-   Free
-   Easy to use
-   Useful! &nbsp;

# Setup

1.  On your machine, run `git clone https://github.com/swxk19/ip.git`
2.  At the <ins>project root</ins>, build the project by running&nbsp;`./gradlew build`
3.  Run the program with&nbsp;`java -jar build/libs/ip.jar`

# Development

### Progress:

- [x] Week 1
- [x] Week 2
- [x] Week 3
- [x] Week 4
- [ ] Week 5
- [ ] Week 6

Automated tests are still under development, send me a PR if you wish to help out with any of them!
<br>I plan to have tests for all methods in all classes.

An example of a test should look like this:

```
    @Test 
    public void toString_dueDateNotDateObject_useUserInputStringFromTo() {
        assertEquals("[D][ ] test deadline (by: non-date byDate)"
                , new Deadline("test deadline", "non-date byDate", false).toString());
    }
```

Commit:
Add Gradle support

swxk19 committed:
Greet user, echo user commands & exit

swxk19 committed:
Close Scanner object after exit command

swxk19 committed:
Add feature to add items to list, and enumerate them on command

swxk19 committed:
Add feature to mark tasks as done or not done

swxk19 committed:
Fix output indentations to match the given examples

swxk19 committed:
Add support for 3 types of tasks: ToDo, Event, Deadline

swxk19 committed:
Error handling for incorrect initialization input of tasks

swxk19 committed:
Handle edge case for error handling in 'Deadline' initialization

swxk19 committed:
Add feature to remove tasks

swxk19 committed:
Implement enums to check valid user commands

swxk19 committed:
Add automated text ui testing

swxk19 committed:
Remove trailing white space in task names of deadlines and events

swxk19 committed:
Remove unused variable 'taskCreationCommands' (replaced to used enums)

swxk19 committed:
Split all inner classes into their own files

swxk19 committed:
Change all classes to public

swxk19 committed:
Duke class: Change if-else statements for to switch-case

swxk19 committed:
Fix issue of not reading next command from user

swxk19 committed:
Move checking of recognizable commands to 'Command' class

swxk19 committed:
Fix missing new line before new user input

swxk19 committed:
Abstract out task list into 'TaskList' class

swxk19 committed:
Fix file paths for test scripts

swxk19 committed:
Abstract out methods for displaying output into 'Ui' class

swxk19 committed:
Rename 'Command' class to 'CommandInput'

swxk19 committed:
Abstract out command executions as a class 'Command'

swxk19 committed:
Abstract 'Mark' command as 'MarkCmd' class

swxk19 committed:
Abstract out 'Unmark' command & fix wrong index being marked/unmarked

swxk19 committed:
Add feature to save task list data on exit

swxk19 committed:
Implements loading tasks from file as string

swxk19 committed:
Abstract out 'todo' command to 'ToDo' class

swxk19 committed:
Abstract out 'event' and 'deadline' commands

swxk19 committed:
Integrate abstracted task commands into main

swxk19 committed:
Abstract out parsing of command inputs into their own methods for task creation

swxk19 committed:
Fix incorrect task name & white spacing for task creation

swxk19 committed:
Add feature to load save tasks from file

swxk19 committed:
Integrate abstract command classes for 'todo', 'event', 'deadline' into main

swxk19 committed:
Remove unused class file

swxk19 committed:
Allow 'deadline' tasks to recognize dates/times

swxk19 committed:
Allow 'event' tasks to recognize dates/times

swxk19 committed:
Change date format output for 'event'

swxk19 committed:
merge branch-Level-8

swxk19 committed:
Fix crashing if loading task list from empty data file

swxk19 committed:
Abstract out methods that load/save tasks from file into the class 'Storage'

swxk19 committed:
Fix incorrect output 'toDate' for 'event' tasks

swxk19 committed:
Abstract out 'delete' command into its own class (Closes Issue #5)

swxk19 committed:
Move parsing for 'mark' command into 'Parser' class

swxk19 committed:
Move parsing for 'unmark' command into 'Parser' class

swxk19 committed:
Move parsing for 'delete' command into 'Parser' class

swxk19 committed:
Change behaviour of 'Command' constructor to take lineInput

swxk19 committed:
Abstract out UI actions into 'Ui' class

swxk19 committed:
Create method 'displayNumTask' in 'Ui' class to display number of tasks in lists

swxk19 committed:
Move parsing for 'deadline' command to 'Parser' class

swxk19 committed:
Move parsing for 'event' command to 'Parser' class

swxk19 committed:
Move parsing for 'todo' command to 'Parser' class

swxk19 committed:
Move parsing in 'Storage' to 'Parser' class

swxk19 committed:
Fix 'todo' not showing proper reply message

swxk19 committed:
Fix needing to type 'bye' twice to exit

swxk19 committed:
Fix extra closing paranthesis being added during task loading

swxk19 committed:
Fix extra border wrapping when displaying number of tasks in list

swxk19 committed:
Package files according to category

swxk19 committed:
Set all fields in classes of /commands to private/protected

swxk19 committed:
Set all fields in classes of /tasks to private/protected

swxk19 committed:
Add newline after 'package' statement in /exception files

swxk19 committed:
Set file path as static variable in 'Storage' class

swxk19 committed:
Move parsing of CommandInput type to 'Parser' class

swxk19 committed:
Add JavaDoc for all /commands files

swxk19 committed:
Add JavaDoc for all /exception files

swxk19 committed:
Reduce verbosity of JavaDoc in /command files

swxk19 committed:
Add JavaDoc for /duke files

swxk19 committed:
Add JavaDoc for all /tasks files

swxk19 committed:
Change name of outputList() method to toString()

swxk19 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

swxk19 committed:
Modify file paths for runtest.sh

swxk19 committed:
Fix handling for EOF without exit command

swxk19 committed:
Setup gradle

swxk19 committed:
Enforce proper line wrapping for line lengths > 120

swxk19 committed:
Add curly brackets to single line conditionals

swxk19 committed:
Add line wrap for long line in 'Deadline' class

swxk19 committed:
Add feature to search for tasks containing a specified keyword

swxk19 committed:
Merge branch 'branch-A-JavaDoc'

swxk19 committed:
Merge branch 'branch-A-CodingStandard'

swxk19 committed:
Merge branch 'branch-Level-9'

swxk19 committed:
Change use of deprecated function 'outputList' to 'toString' in 'FindCmd' class

swxk19 committed:
Remove debugging print statements in 'TaskList'

swxk19 committed:
Fix runtime error if missing info in user input (Closes Issues #2 & #10)

swxk19 committed:
Add JUnit tests for 'TaskList', 'ToDo', 'Event', 'Deadline' classes

swxk19 committed:
Remove class files from repo (.class files should not be version-controlled)

SeanChinJunKai commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,78 @@
+    private static String fileName = "../../../data/duke.txt";
+
+    /**
+     * Saves a task list to a file.
```
Should you have empty lines between description and parameter section for JavaDocs?

SeanChinJunKai commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,70 @@
+package duke;
```
I like how you have organised the exceptions, commands and tasks into their own sub packages of duke.

SeanChinJunKai commented on src/main/java/duke/Ui.java:
```diff
@@ -0,0 +1,91 @@
+public class Ui {
+    private static Scanner sc = new Scanner(System.in);
+
+    // For checking whether to continue running Duke, i.e. exit command has not been called.
```
Would it be a good idea to put the comments for class members as JavaDocs. For example

/** Number of connections to this database */
private int connectionCount;

SeanChinJunKai commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,199 @@
+
+    /**
+     * Used by 'CommandInput' class to get CommandInput type from command input.
+     * @param lineInput Command line input that user entered
```
Should you include punctuation behind each parameter description?

SeanChinJunKai reviewed (commented):
LGTM. Just some nits to fix.

swxk19 committed:
Refactor Duke class to follow Week 3's example (Closes Issue #13)

swxk19 committed:
Move command switch-case from Duke::main to CommandInput::getCommandFromInput

swxk19 committed:
Add checkstyle configurations

swxk19 committed:
Fix checkstyle errors in 'CommandInput'

swxk19 committed:
Fix checkstyle for 'Parser'

swxk19 committed:
Fix checkstyle errors for 'Storage'

swxk19 committed:
Fix checkstyle errors for 'Ui'

swxk19 committed:
Fix checkstyle errors for 'Deadline'

swxk19 committed:
Fix checkstyle errors for 'Event'

swxk19 committed:
Fix checkstyle errors for 'Task'

swxk19 committed:
Fix checkstyle errors for 'Storage'

swxk19 committed:
Fix checkstyle errors for 'ByeCmd'

swxk19 committed:
Fix checkstyle errors for 'DeadlineCmd'

swxk19 committed:
Fix checkstyle errors for 'FindCmd'

swxk19 committed:
Fix checkstyle errors in all files

swxk19 committed:
Init JavaFX according to tutorial

swxk19 committed:
Implement base working model of GUI

swxk19 committed:
Implement TaskList::find to use varargs

swxk19 committed:
Merge branch 'branch-A-Varargs'

swxk19 committed:
Refactor where Duke::sendResponse is called to be in Duke

swxk19 committed:
Fix Duke not greeting user on start

swxk19 committed:
Move stage actions into 'StageHandler' class

swxk19 committed:
Fix checkstyles

swxk19 committed:
Add assertions to check that commands are properly executed

Ensures that the size of the list should grow/shrink if tasks are
added/deleted.

Ensures that tasks completion status are modified when mark/unmark
is called.

swxk19 committed:
Setup GitHub Actions

swxk19 committed:
Use String.format() to construct duke responses

swxk19 committed:
Use String.format() for duke responses in duke.tasks

Closes Issue #15

Previously, '+' operator was used to join strings to construct a
duke response, this may be difficult to read. Using String.format()
has better readability.

swxk19 committed:
Abstract 'ParsedLoadedTask' into its own file

Move and renamed 'ParsedLoadedTask' to its own file 'ParsedTask',
which stores the relevant information of a task before its actual creation.
Currently, it is only used by 'Storage' when loading tasks from a saved file,
but should be later generalised to parse user commands as well.
(as opposed to current method of storing parsed info in an array)

swxk19 committed:
Change duke response to return from Command::execute()

swxk19 committed:
Modify TaskList::find to use streams instead of nested for-loop

swxk19 committed:
Abstract init of gui specifications into its own file

swxk19 committed:
Split initialization of components into their own methods

swxk19 committed:
Fix typo in Event::stringFields

swxk19 committed:
Merge pull request #16 from swxk19/branch-A-CodeQuality

Improves code quality.

swxk19 committed:
Merge branch 'master' into branch-A-CI

swxk19 committed:
Merge branch 'master' into branch-A-Assertions

swxk19 committed:
Merge pull request #18 from swxk19/branch-A-Assertions

Use Assertions

swxk19 committed:
Merge branch 'master' into branch-A-CI

swxk19 committed:
Merge pull request #20 from swxk19/branch-A-CI

Set up CI

swxk19 committed:
Add feature to view ongoing tasks for a specified date

swxk19 committed:
Fix file paths for loading/saving from file

swxk19 committed:
Update .gitignore to ignore data file

swxk19 committed:
Refactor loading of tasks in Storage::loadFromFile

swxk19 committed:
Refactor to follow JavaFX tutorial part 4

swxk19 committed:
Complete exercises for JavaFX tutorial part 4 (Closes #Issue 21)

swxk19 committed:
Remove dead code

swxk19 committed:
Add newline after method descriptor in all JavaDoc headers

swxk19 committed:
Add custom colors for Duke response

swxk19 committed:
Add border radius to Duke response

swxk19 committed:
Add invalid command error highlighting

swxk19 committed:
Allow DialogBox to have flexible height, if message is long

swxk19 committed:
Add more colors

swxk19 committed:
Remove background of profile images

swxk19 committed:
Add owl personality to duke

swxk19 committed:
Changes 'bye' command execution to also close application

swxk19 committed:
Add User Guide

swxk19 committed:
Change Owlly response for 'view' command

swxk19 committed:
Add padding for dialog text

swxk19 committed:
Disable resizing

swxk19 committed:
Fix checkstyle

swxk19 committed:
Fix crash for deleting tasks out of range

Closes Issue #6

swxk19 committed:
Rename var DialogBox:dialogBoxStyle

swxk19 committed:
Change catch exception type in TaskList::removeTask

swxk19 committed:
Allow saving on list modification

swxk19 committed:
Add Ui image

swxk19 committed:
Change TaskList::toString() output

swxk19 committed:
Allow resizing of window

swxk19 committed:
Fix newlines being added to data file

swxk19 committed:
Rename Ui class to MessageFormat

swxk19 committed:
Add and fix JavaDoc for all methods & classes

swxk19 committed:
Add response for if list is empty for list command

Closes Issue #1

swxk19 committed:
Fix JavaDocs

swxk19 committed:
Add message to inform user if tasks were loaded from a saved file

Closes Issue #14

swxk19 committed:
Add error handling for missing date in view command

swxk19 committed:
Fix 'view' command crashing

swxk19 committed:
Fix incorrect behaviour of 'unmark'

swxk19 committed:
Change response for 'view' command

swxk19 committed:
Fix checkstyle

swxk19 committed:
Reformat UG & add section for setting up

swxk19 committed:
Update README.md

swxk19 committed:
Update README.md

swxk19 committed:
Update README.md
